spec/javascripts/govuk_publishing_components/analytics-ga4/ga4-search-tracker.spec.js

```diff
@@ -0,0 +1,136 @@
+/* eslint-env jasmine */
+
+describe('Google Analytics search tracking', () => {
+  'use strict'
+
+  let fixture, form, input, ga4SearchTracker
+  const GOVUK = window.GOVUK
+
+  const html = `
+    <form
+      data-module="ga4-search-tracker"
+      data-ga4-search-type="site search"
+      data-ga4-search-url="/search"
+      data-ga4-search-section="section"
+      data-ga4-search-index-section="19"
+      data-ga4-search-index-section-count="89"
+    >
+      <input type="search" name="keyword" value="initial value">
+      <button type="submit">Search</button>
+    </form>
+  `
+
+  beforeAll(() => {
+    GOVUK.analyticsGa4 = GOVUK.analyticsGa4 || {}
+    GOVUK.analyticsGa4.vars = GOVUK.analyticsGa4.vars || {}
+    GOVUK.analyticsGa4.vars.gem_version = 'aVersion'
+  })
+
+  afterAll(() => {
+    window.dataLayer = []
+  })
+
+  beforeEach(() => {
+    fixture = document.createElement('div')
+    document.body.appendChild(fixture)
+    fixture.innerHTML = html
+
+    form = fixture.querySelector('form')
+    input = form.querySelector('input')
+
+    spyOn(GOVUK.analyticsGa4.core, 'applySchemaAndSendData')
+
+    ga4SearchTracker = new GOVUK.Modules.Ga4SearchTracker(form)
+  })
+
+  afterEach(() => {
+    fixture.remove()
+  })
+
+  describe('if the search field is missing', () => {
+    beforeEach(() => {
+      GOVUK.setConsentCookie({ usage: true })
+
+      form.removeChild(input)
+      ga4SearchTracker = new GOVUK.Modules.Ga4SearchTracker(form)
+      ga4SearchTracker.init()
+    })
+
+    it('does not track search events', () => {
+      GOVUK.triggerEvent(form, 'submit')
+
+      expect(GOVUK.analyticsGa4.core.applySchemaAndSendData).not.toHaveBeenCalled()
```

Comment:
Is this able to be more succinct at all? as it's quite a mouthful.

Not sure if we can do: 

```
const applySchemaAndSendDataSpy = spyOn(GOVUK.analyticsGa4.core, 'applySchemaAndSendData')
expect(applySchemaAndSendDataSpy).not.toHaveBeenCalled()
```